day2OfJS/day2.js

```diff
@@ -0,0 +1,234 @@
+const randomString = 'JavaScript';
+const random = Math.floor(Math.random() * randomString.length);
+console.log(random); //8
+
+
+/*
+11. Use console.log() and escape characters to print the following pattern.
+1 1 1 1 1
+2 1 2 4 8
+3 1 3 9 27
+4 1 4 16 64
+5 1 5 25 125
+*/
+console.log('1\t1\t1\t1\t1\n2\t1\t2\t4\t8\n3\t1\t3\t9\t27\n4\t1\t4\t16\t64\n5\t1\t5\t25\t125');
+/*
+1       1       1       1       1
+2       1       2       4       8
+3       1       3       9       27
+4       1       4       16      64
+5       1       5       25      125
+*/
+
+
+//12. Use substr to slice out the phrase 'You cannot end a sentence with because because because is a conjunction' from the following sentence:'You cannot end a sentence with because because because is a conjunction'
+const phrase4 = 'You cannot end a sentence with because because because is a conjunction';
+console.log(phrase4.substr(31, 24)); //because because because
+
+
+
+
+//Exercises: Level 3
+//1. 'Love is the best thing in this world. Some found their love and some are still looking for their love.' Count the number of word love in this sentence.
+const love = 'Love is the best thing in this world. Some found their love and some are still looking for their love.';
+console.log(love.match(/love/gi)); //[ 'Love', 'love', 'love' ]
+
+
+//2. Use match() to count the number of all because in the following sentence:'You cannot end a sentence with because because because is a conjunction'
+const because = 'You cannot end a sentence with because because because is a conjunction';
+console.log(because.match(/because/gi)); //[ 'because', 'because', 'because' ]
+
+
+//3. Clean the following text and find the most frequent word(hint, use replace and regular expressions).
+const sentence = '%I $am@% a %tea@cher%, &and& I lo%#ve %te@a@ching%;. The@re $is no@th@ing; &as& mo@re rewarding as educa@ting &and& @emp%o@weri@ng peo@ple. ;I found tea@ching m%o@re interesting tha@n any ot#her %jo@bs./%Do@es thi%s mo@tiv#ate yo@u to be a tea@cher!? %Th#is 30#Days&OfJavaScript &is al@so $the $resu@lt of &love& of tea&ching';
+console.log(sentence.replace(/[^a-zA-Z 0-9.]+/gi, ''));
+/*I am a teacher and I love teaching. There is nothing as more rewarding as educating and empowering people. I found teaching more interesting than any other jobs.
+Does this motivate you to be a teacher This 30DaysOfJavaScript is also the result of love of teaching
+*/
+
+
+//4. Calculate the person's total annual income by extracting the figures from the following text. 
+//'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month'.
+const salary = 'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month';
+const money = salary.match(/(\d+)/g);
+let multiplication1 = Number(money[0]) * 12;
+let multiplication2 = Number(money[1]);
+let multiplication3 = Number(money[2]) * 12;
```

Comment:
Remember to always use `const` unless you are going to change the value of the variable later on.